File: src/java/search/SequentialSearch.java
package search;

public class SequentialSearch {

    public static final int WILDCARD_VALUE = -1;

    public static int sequentialSearch(int[] array, int value) {

        for (int i = 0; i < array.length; i++) {
            if (value == array[i]) {
                return i;
            }
        }
        return WILDCARD_VALUE;
    }
}

File: src/c++/search/sequentialSearch.cpp
#include <stdio.h>
#define MAX 100

int sequential_search(int n, int v[MAX], int x) {

    int k;

    for (k = n -1; k >= 0 && v[k] != x; k--)
        ;
    return k;
}

// Recursive
int sequential_search_recursive(int n, int v[MAX], int x) {

    if (n==0) {
        return -1;
    } else if (x == v[n -1]) {
        return n - 1;
    } else {
        return sequential_search_recursive(n - 1, v, x);
    }
}